pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -849,17 +848,22 @@ public void takeSnapshot() {
       _logger.info("Skip taking snapshot because metadata manager is already stopped");
       return;
     }
-    _snapshotLock.writeLock().lock();
+
     try {
       long startTime = System.currentTimeMillis();
+      while (!_snapshotLock.writeLock().tryLock(5, TimeUnit.MINUTES)) {
+        _logger.warn("Unable to acquire snapshotLock.writeLock in: {}", System.currentTimeMillis() - startTime);
+      }
```

Comment:
Let's move this outside of the `try` block, then we don't need other changes, and this is the best practice

Reply:
Actually we want to track locking time and snapshot time separately